linkedlist.ts

```diff
@@ -34,8 +34,42 @@ export default class LinkedList<V> {
     return value;
   }
 
+  filter(filter: (item: V) => boolean) {
```

Comment:
This mutates the instance it is called upon. I would suggest keeping this aligned with `Array#filter()`, lodash/underscore's `.filter()` and most implementation of the same concept (insofar as my experience goes) by making it return a new instance. Or, renaming to something different that makes it clear we're mutating the original one.